app/models/contract.rb

```diff
@@ -326,7 +326,7 @@ def intersect_with_unpaid_period(period, paid_end)
     from = [ period.begin.to_date, paid_end ].max.to_date
     to = [ period.end.to_date, from ].max.to_date
 
-    from.to_time..to.to_time.end_of_day
+    from.to_time(:utc)..to.to_time(:utc).end_of_day
```

Comment:
@guicassolato why do we need to use `utc` now?

Reply:
That's a tricky question :) For this PR, no. So removing. Thanks for pointing up.

But we sure need to discuss more about timezones in billing. Methods like `to_time` and `to_date` are dangerous, specially in porta where UTC and provider-specific TZ are both used over the same data set.